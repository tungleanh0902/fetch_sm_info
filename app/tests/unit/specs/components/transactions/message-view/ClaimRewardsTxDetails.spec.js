import { shallowMount } from "@vue/test-utils"
import ClaimRewardsTxDetails from "src/components/transactions/message-view/ClaimRewardsTxDetails"

describe(`ClaimRewardsTxDetails`, () => {
  let wrapper, $store

  $store = {
    getters: {
      isExtension: false,
      network: `cosmos-hub-testnet`,
    },
  }

  const tx = {
    type: "ClaimRewardsTx",
    hash: "6D33991B1150161AF8D839B165E8EE84377213142D1FC1091FF4BAA783114678",
    height: 933171,
    timestamp: "2020-02-20T10:43:36Z",
    memo: "",
    success: true,
    fees: [],
    details: {
      from: ["cosmosvaloper123"],
      amounts: [
        {
          denom: "ATOM",
          amount: "10",
        },
      ],
    },
  }

  const multiTx = {
    type: "ClaimRewardsTx",
    hash: "6D33991B1150161AF8D839B165E8EE84377213142D1FC1091FF4BAA783114678",
    height: 933171,
    timestamp: "2020-02-20T10:43:36Z",
    memo: "",
    success: true,
    fees: [],
    details: {
      from: ["cosmosvaloper123", "cosmosvaloper456", "cosmosvaloper789"],
      amounts: [
        {
          denom: "ATOM",
          amount: "3",
        },
      ],
    },
  }

  const validators = {
    cosmosvaloper123: {
      name: `SuperValidator`,
      operatorAddress: `cosmosvaloper123`,
      picture: "",
      __typename: `Validator`,
    },
    cosmosvaloper456: {
      name: `AwesomeValidator`,
      operatorAddress: `cosmosvaloper456`,
      picture: "",
      __typename: `Validator`,
    },
    cosmosvaloper789: {
      name: `GiveMeYourMoney`,
      operatorAddress: `cosmosvaloper789`,
      picture: "",
      __typename: `Validator`,
    },
  }

  it(`renders a claim rewards transaction message`, () => {
    wrapper = shallowMount(ClaimRewardsTxDetails, {
      propsData: {
        transaction: tx,
        validators: validators,
        show: false,
      },
      mocks: {
        $store,
      },
      stubs: [`router-link`],
    })
    expect(wrapper.element).toMatchSnapshot()
  })

  it(`renders a multi claim rewards transaction message`, () => {
    wrapper = shallowMount(ClaimRewardsTxDetails, {
      propsData: {
        transaction: multiTx,
        validators: validators,
        show: false,
      },
      mocks: {
        $store,
      },
      stubs: [`router-link`],
    })
    expect(wrapper.element).toMatchSnapshot()
  })

  it(`returns an empty array if validators haven't been loaded yet`, () => {
    const self = {
      validators: {},
    }
    const getValidatorsResponse = ClaimRewardsTxDetails.computed.getValidators.call(
      self
    )
    expect(getValidatorsResponse).toEqual([])
  })
})
