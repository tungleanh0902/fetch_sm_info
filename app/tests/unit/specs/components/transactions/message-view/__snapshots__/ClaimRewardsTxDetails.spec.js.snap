// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClaimRewardsTxDetails renders a claim rewards transaction message 1`] = `
<div
  class="claim-rewards-wrapper"
>
  <div
    class="tx__content"
  >
    <transactionicon-stub
      transactiontype="Claimed"
    />
     
    <div
      class="tx__content__left"
    >
      <h3>
        Claimed
      </h3>
       
      <div
        class="validator-row"
      >
        <span>
          From
        </span>
         
        <router-link-stub
          class="validator-link"
          to="[object Object]"
        >
          <avatar-stub
            address="cosmosvaloper123"
            alt="generic validator logo - generated avatar from address"
            class="validator-image"
          />
        </router-link-stub>
      </div>
    </div>
  </div>
   
  <div
    class="tx__content__right"
  >
    <p
      class="amount"
    >
      
        10  ATOM
      
    </p>
  </div>
</div>
`;

exports[`ClaimRewardsTxDetails renders a multi claim rewards transaction message 1`] = `
<div
  class="claim-rewards-wrapper"
>
  <div
    class="tx__content"
  >
    <transactionicon-stub
      transactiontype="Claimed"
    />
     
    <div
      class="tx__content__left"
    >
      <h3>
        Claimed
      </h3>
       
      <div
        class="validator-row"
      >
        <span>
          From
        </span>
         
        <router-link-stub
          class="validator-link"
          to="[object Object]"
        >
          <avatar-stub
            address="cosmosvaloper123"
            alt="generic validator logo - generated avatar from address"
            class="validator-image"
          />
        </router-link-stub>
        <router-link-stub
          class="validator-link"
          to="[object Object]"
        >
          <avatar-stub
            address="cosmosvaloper456"
            alt="generic validator logo - generated avatar from address"
            class="validator-image"
          />
        </router-link-stub>
        <router-link-stub
          class="validator-link"
          to="[object Object]"
        >
          <avatar-stub
            address="cosmosvaloper789"
            alt="generic validator logo - generated avatar from address"
            class="validator-image"
          />
        </router-link-stub>
      </div>
    </div>
  </div>
   
  <div
    class="tx__content__right"
  >
    <p
      class="amount"
    >
      
        3  ATOM
      
    </p>
  </div>
</div>
`;
